integreat_compass/cms/views/interactions/vote_form_view.py

```diff
@@ -0,0 +1,110 @@
+        offer_versions = (
+            OfferVersion.objects.select_related("offer__organization")
+            .prefetch_related("offer__tags", "votes")
+            .annotate(
+                approves=Count(Case(When(votes__approval=True, then=1))),
+                declines=Count(Case(When(votes__approval=False, then=1))),
+            )
+            .filter(pk__in=latest_version_of_offer)
+        )
+        pending_offer_versions = list(
+            filter(
+                lambda offer_version: offer_version.state
+                == offer_version_states.PENDING,
+                offer_versions,
+            )
+        )
+        for version in pending_offer_versions:
+            version.user_vote = version.votes.filter(creator=request.user).first()
+            version.pending = (
+                version.number_of_votes_needed - version.approves - version.declines
+            )
+        return pending_offer_versions
+
+    def get_context_data(self, *args, **kwargs):
+        r"""
+        Render :class:`~integreat_compass.cms.forms.interactions.vote_form.VoteForm`,
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param \**kwargs: The supplied keyword arguments
+        :type \**kwargs: dict
+
+        :raises ~django.core.exceptions.PermissionDenied: If user does not have the permission to edit the specific page
+
+        :return: The rendered template response
+        :rtype: ~django.template.response.TemplateResponse
+        """
+        context = super().get_context_data(**kwargs)
+        pending_offer_versions: list = self.get_pending_offer_versions(self.request)
+        context.update({"pending_offer_versions": pending_offer_versions})
+        return context
+
+    @transaction.atomic
+    def post(self, request, *args, **kwargs):
+        r"""
+        Add vote of a user to an offer version
+
+        :param self: Object representing the user call
+        :type self: ~django.http.HttpRequest
+
+        :param request: The current request
+        :type request: ~django.http.HttpRequest
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param \**kwargs: The supplied keyword arguments
+        :type \**kwargs: dict
+
+        :return: The rendered template response
+        :rtype: ~django.template.response.TemplateResponse
+        """
+        offer_version_id = request.POST.get("offer_version_id")
+        approval = request.POST.get("approval")
+        Vote.objects.update_or_create(
+            creator=request.user,
+            offer_version_id=offer_version_id,
+            defaults={"approval": approval},
+        )
+        messages.add_message(
+            request,
+            messages.SUCCESS,
+            _('Your vote on "{}" has successfully been submitted.').format(
+                OfferVersion.objects.get(id=offer_version_id).title
+            ),
+        )
+        return redirect("/applications/")
```

Comment:
```suggestion
        return redirect("cms:protected:votes")
```
The only place where you use hardcoded paths is the url config. In all other places, names url patterns are preferred.